### ISSUE TITLE
Keyboard Navigation

### ISSUE THREAD

**Issue by smeragoel** [Quansight-Labs/czi-scientific-python-mgmt#56]
## ⌨️ Keyboard navigation
Users must be able to _navigate to_ all interactive interface components using a keyboard - [WCAG 2.1.1](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html)

```[tasklist]
### Tasks
- [x] User cannot navigate into code blocks / snippets  https://github.com/pydata/pydata-sphinx-theme/pull/1636
- [x] Cannot navigate to section "chevron" dropdown expander/collapser ![image](https://github.com/Quansight-Labs/czi-scientific-python-mgmt/assets/98317216/687026b8-6b43-4f9e-8b1f-cec58aa7317d) - fixed by https://github.com/pydata/pydata-sphinx-theme/pull/1564
- [x] Copy buttons do not have focus indicator (potentially fixed by https://github.com/pydata/pydata-sphinx-theme/pull/1564)
- [ ] Read the docs version switcher _(note: this may have been fixed since I cannot reproduce the problem in the latest version, but I am adding it here to cross-check with the team) - need issue in https://github.com/readthedocs/readthedocs.org/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+version
```
## 🪤 No keyboard traps 
Users must be able to _navigate away_ from all components using a keyboard - [WCAG 2.1.2](https://www.w3.org/WAI/WCAG21/Understanding/no-keyboard-trap.html)

```[tasklist]
### Tasks
- [ ] Search Box Modal (Full Screen)
```

**Comment by gabalafou** [comment]
I've noticed that the Read the Docs switcher doesn't always consistently load the same way. Sometimes it appears as an overlay in the bottom right, other times it appears as the last item in the primary (left) sidebar. 

Maybe that's why you're unable reproduce the keyboard issue?
